client/packages/common/src/utils/types/TypeUtils.ts

```diff
@@ -1,5 +1,5 @@
 export const TypeUtils = {
-  isTypeOf: <T>(
+  isTypeOf: <T extends object>(
```

Comment:
Not strictly related, but should this method also check that the `variableToCheck` *is* an object before the `in` check? Then it doesn't need the `extends` as `T` could be any type.
```
typeof variableToCheck === 'object' &&
    variableToCheck !== null &&
    field in variableToCheck,
```

Also, this method doesn't actually match the type, it just checks that the property exists. The property itself could still be of a different type to the expected value, so this is somewhat misleadingly named. (`hasProperty` perhaps?). I'm actually not sure what this method does that is any different to just doing `<field> in variable` in place. (It's only used once as far as I can tell, and in that case using `if "expiryDate" in rowData` seems to do exactly the same thing and is arguably more straightforward). 

Do we need it? 

Sorry, bit of a tangent -- feel free to ignore 😆

Reply:
ideally that `isTypeOf` would be checking both the existence of the property and that it is the correct type. But since it doesn't.. I've removed. I think we have the same functionality with less code and more clarity with the change. Thanks!